exercises/concept/himaris-cake-shop/.meta/exemplar.py

```diff
@@ -0,0 +1,23 @@
+# For 1kg cake
+BASE_PRICE = 200
+TIME_TO_BAKE_BASE = 40
+
+def calculate_base_price(weight_of_base, number_of_layers): # Positional Args
+    return BASE_PRICE * weight_of_base * number_of_layers
+
+def calculate_time_for_baking_base(weight_of_base, number_of_layers=1): # Deafult Args/Keyword Args
+    return TIME_TO_BAKE_BASE * weight_of_base * number_of_layers
+
+def total_ingredients_price(*ingredients_prices): # *args method
+    PRICE = 0
+    for price in ingredients_prices:
+        PRICE += price
+    return PRICE
+
+def calculate_total_price(weight_of_base=1, number_of_layers=1, **ingredients_prices): # *kwargs method
+    return (calculate_base_price(weight_of_base) * number_of_layers) + total_ingredients_price(*ingredients_prices.values())
+
+def make_cake():
+    # Don't know what to put in here yet or whether we need this
+    pass
```

Comment:
any ideas to implement this? @BethanyG 
and I also wanted to know if the sequencing of functions is correct according to sequence of learning, do we need any addition her?

Reply:
Hi @asheeeshh 

Sorry for the time it's taken to get back to you.  😞   I was playing with this this weekend.  I think it's a good _start_, but the scenario felt too simple to me, so I was trying to draft something a little more involved/detailed.  Something that would demonstrate _why_ you would pass arbitrary args in one form or another. But I might have made a mess of it.   I do think that the order we present the types is correct:

 1. positional
 2. positional + named/default, 
 3. positional + default + arbitrary
 4. positional + default + arbitrarily named
 5. _Didn't come up with something for this (yet!)_

I am going to spend some more time today cleaning up what I did this weekend, and when I am done I will post it here to see what you think.  I might also come back and adjust that pricing list (_should we do it in Yen??_).

So far, I have the following story and "pricing", but am trying to work out how those things would now map to tasks:

## Story

Himari is opening a new custom cake shop, and has asked for your help with her costing, online ordering, and POS (Point of Sale) systems.

She is really into flavors, fillings, and decorations, so has decided to focus on just two sizes: 15cm/6in and 20cm/8in rounds. 

Cakes are priced by their **weight** (_starting with a base weight of aprox 1kg, and going up by 1/2kg for each layer added_). Each 20cm layer is approximately 1/2kg (_she divides the thicker single cake in half._). Each 15cm layer is _also_ 1/2kg (_15cm layers are thicker than 20cm layers_). 

There are additional charges for each custom frosting, filling, or topping that is added to a cake order. There is also a surcharge for bakes that take a longer amount of time (_cakes over 4 layers_) to cook or to decorate.

Each basic 2-layer cake is filled with vanilla custard and frosted with fresh whipped cream.  While she does offer a 1kg single-layer 20cm cake, it comes with the filling or custard on the side - so there is no filling discount for a single-layer cake. The pricing for base filling and frosting is included in the price for each added tier.

**Note:** Himari's oven can fit up to 6 15cm/6in cake pans and 4 20cm/8in cake pans at a time, and making decorations and various fillings and frostings in bulk also saves time --  but we're keeping bake times and decoration times linear for simplicity of calculation.

Himari's pricing and timing tables are below:

| **Item**                                                                                                                                                	| **Price** 	|
|---------------------------------------------------------------------------------------------------------------------------------------------------------	|-----------	|
| Small Size Discount (_for customizing cakes that are 15cm_)                                                                                             	| 0.85      	|
| Base Price (_1kg - 1.3kg of cake_)<br> This is a 1 or 2 layer 20cm cake or a 2 layer 15cm cake.                                                                                                                    	| 200       	|
| Tier Price (_.5kg - .65kg of cake_)<br> This is 1 layer of a 20cm or 15cm cake.                                                                                                                     	| 170       	|
| Customization Surcharge                                                                                                                                 	| 5         	|
| Chocolate Ganache Frosting                                                                                                                              	| 55        	|
| Flavored Buttercream Frosting (_vegan and vegetarian available_)<br>Vanilla, Chocolate, Raspberry, Blackberry, Grapefruit, Orange, Lemon, Coconut, Lime 	| 50        	|
| Flavored Whipped Cream Frosting (_vegan & vegetarian available_)<br>Vanilla, Chocolate, Raspberry, Blackberry, Grapefruit, Orange, Lemon, Coconut, Lime 	| 40        	|
| Fruit Glaze                                                                                                                                             	| 30        	|
| Honey Glaze                                                                                                                                             	| 30        	|
| Flavored Custard (_vegan & vegetarian available_)<br>Coconut, Hazelnut, Pistachio, or Sesame                                                            	| 30        	|
| Chocolate Ganache Filling                                                                                                                               	| 50        	|
| Nuts Mixin <br>Coconut, Hazelnut, Pistachio, or Sesame                                                                                                  	| 40        	|
| Fruit Mixin <br>Strawberries, Blackberries, Raspberries, Gooseberries, Grapes, Melon, or Citrus                                                         	| 50        	|
| Fruit Filling (_replaces custard with fruit_)<br>Strawberries, Blackberries, Raspberries, Gooseberries, Grapes, Melon, or Citrus                        	| 5         	|
| Jam Filling (_replaces custard with Jam_)<br>Strawberry, Blackberry, Raspberry, Marmalade, or Grape                                                     	| 5         	|
| Berry Topping <br>Strawberries, Blackberries, Raspberries, Gooseberries                                                                                 	| 50        	|
| Sliced Fruit Topping <br>Mangoes, Oranges, Melon, Grapes, Kiwi, Banana                                                                                  	| 40        	|
| Mixed Fruit Topping                                                                                                                                     	| 30        	|
| Nut Topping <br>Coconut, Hazelnut, Pistachio, or Sesame                                                                                                 	| 25        	|

<br><br>

| **Item**              	| **Time to Make** 	|
|-----------------------	|------------------	|
| Time to Bake Base     	| 40               	|
| Time to Bake Tier     	| 20               	|
| Time to Decorate Base 	| 15               	|
| Time to Decorate Tier 	| 5                	|